scipy/io/_fortran.py

```diff
@@ -133,7 +133,7 @@ def _read_size(self, eof_ok=False):
         elif len(b) < n:
             raise FortranFormattingError(
                 "End of file in the middle of the record size")
-        return int(np.frombuffer(b, dtype=self._header_dtype, count=1))
+        return np.frombuffer(b, dtype=self._header_dtype, count=1)[0]
```

Comment:
Perhaps wise to leave the `int` here in case consumers are not expecting a numpy type

Reply:
Added the `int` and left the `np.buffer` + `count=1`